### PR TITLE
Add regression test for issue#10447

### LINKED ISSUE #10447: Query: Navigation property populated twice
Hey.
It seems that in some instances it's possible for an `Include`ed navigation list to receive duplicate items when it's being queried while its dbset is being iterated on.

Now, I'm no EF expert so maybe it's a misunderstanding on my part but I'm assuming that querying is always supposed to be a no-op, whether you use `Include()` or not.

To be more explicit, here is a small standalone repro :
```c#
    public class Parent
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        
        public string Name { get; set; }

        public IEnumerable<Child> Children { get; set; }
    }

    public class Child {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public Parent Entry { get; set; }
    }

    public class Ctx : DbContext
    {
        public DbSet<Parent> Parents { get; set; }
        public DbSet<Child> Children { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlite("Data Source=./db;");
        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            // populate
            using (var db = new Ctx()) {
                db.Database.EnsureCreated();
                var p1 = new Parent {
                    Name = "A1",
                    Children = new List<Child> { }
                };
                var p2 = new Parent {
                    Name = "A2",
                    Children = new List<Child> {
                        new Child { },
                    }
                };
                db.Add(p1); db.Add(p2);
                db.SaveChanges();
            }

            // read back
            using (var db = new Ctx()) {
                foreach (var entry in db.Parents
                                        .Include(e => e.Children)) {
                    // should be no-op ?
                    db.Parents
                        .Include(e => e.Children)
                        .ToList();
                }

                foreach (var entry in db.Parents
                                        .Include(e => e.Children)) {
                    Console.WriteLine($"parent {entry.Name} ({entry.Id}) :");
                    foreach (var child in entry.Children) {
                        Console.WriteLine($"  child ({child.Id})");
                    }
                }
            }
        }
    }
```

Which outputs :
```
parent A1 (1) :
parent A2 (2) :
  child (1)
  child (1)
```
EF core version : 2.0.1

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #10447

### CODE REVIEW

File: test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs [hunk 1]
```diff
@@ -6388,6 +6388,93 @@ public class OrganisationUser7973
...
+
+                    Assert.Collection(query,
+                        b => Assert.Equal(3, b.Posts.Count),
+                        b => Assert.Equal(2, b.Posts.Count),
+                        b => Assert.Single(b.Posts));
+                }
+            }
+        }
+
+        public class MyContext10447 : DbContext
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: You don't need to create a new context and entity classes for every test

Comment (smitpatel) [reply test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: I tried to use this in existing but it requires MARS to be on so it ended up being in QueryBugsTest.

Comment (smitpatel) [reply test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: Could reuse same context. I was hoping that a non-shared-model test infra would make this easier in future.

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: You can use an existing context in QueryBugsTest or add a test attribute that indicates that it requires MARS

Comment (smitpatel) [reply test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: I added few points in #17588
There are pros & cons for using existing context. We can discuss design. For now, I will keep adding more context to keep it "consistent".

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: It's only consistent within the file, not within the solution. We should reopen https://github.com/aspnet/EntityFrameworkCore/issues/1015

**Pull request merged**